src/acl/external/time_quota/ext_time_quota_acl.cc

```diff
@@ -325,88 +286,93 @@ static void processActivity(const char *user_key)
+            writeTime(user_key, KeyPeriodStart, periodStart);
+            timeBudgetConfigured = readTime(user_key, KeyTimeBudgetConfigured);
             if ( timeBudgetConfigured == 0 ) {
-                log_debug("No time budget configured for user \"%s\". Quota for this user disabled.\n", user_key);
-                writeTime(user_key, KEY_TIME_BUDGET_LEFT, pauseLength);
+                log_debug("No time budget configured for user \"" << user_key  <<
+                          "\". Quota for this user disabled.");
+                writeTime(user_key, KeyTimeBudgetLeft, pauseLength);
             } else {
-                writeTime(user_key, KEY_TIME_BUDGET_LEFT, timeBudgetConfigured);
+                writeTime(user_key, KeyTimeBudgetLeft, timeBudgetConfigured);
             }
         }
     }
 
     // [2] Decrease time budget iff activity
-    lastActivity = readTime(user_key, KEY_LAST_ACTIVITY);
+    lastActivity = readTime(user_key, KeyLastActivity);
     if ( lastActivity == 0 ) {
         // This is the first request ever
-        writeTime(user_key, KEY_LAST_ACTIVITY, now);
+        writeTime(user_key, KeyLastActivity, now);
     } else {
         activityLength = now - lastActivity;
         if ( activityLength >= pauseLength ) {
             // This is an activity pause.
-            log_debug("Activity pause detected for user \"%s\".\n", user_key);
-            writeTime(user_key, KEY_LAST_ACTIVITY, now);
+            log_debug("Activity pause detected for user \"" << user_key << "\".");
+            writeTime(user_key, KeyLastActivity, now);
         } else {
             // This is real usage.
-            writeTime(user_key, KEY_LAST_ACTIVITY, now);
+            writeTime(user_key, KeyLastActivity, now);
 
-            log_debug("Time budget reduced by %ld for user \"%s\".\n",
-                      activityLength, user_key);
-            timeBudgetCurrent = readTime(user_key, KEY_TIME_BUDGET_LEFT);
+            log_debug("Time budget reduced by " << activityLength <<
+                      " for user \"" << user_key << "\".");
+            timeBudgetCurrent = readTime(user_key, KeyTimeBudgetLeft);
             timeBudgetCurrent -= activityLength;
-            writeTime(user_key, KEY_TIME_BUDGET_LEFT, timeBudgetCurrent);
+            writeTime(user_key, KeyTimeBudgetLeft, timeBudgetCurrent);
         }
     }
 
-    timeBudgetCurrent = readTime(user_key, KEY_TIME_BUDGET_LEFT);
-    snprintf(message, TQ_BUFFERSIZE, "message=\"Remaining quota for '%s' is %d seconds.\"", user_key, (int)timeBudgetCurrent);
+    timeBudgetCurrent = readTime(user_key, KeyTimeBudgetLeft);
+
+    std::ostringstream oss;
+    oss << "message=\"Remaining quota for '" << user_key << "' is " << timeBudgetCurrent << " seconds.\"";
+    auto message = oss.str();
     if ( timeBudgetCurrent > 0 ) {
-        log_debug("OK %s.\n", message);
+        log_debug("OK " << message);
```

Comment:
IMO we do not need to log values which are passed to our `SEND_*()` macros. Squid will cache.log the reply message with `debugs(82, 2, "reply=" ... )` .  Running it manually for testing will see the `SEND_ERR()` stdout content.

Reply:
removed